### PR TITLE
refactor(thirdweb): update siwe interface

### PR DESCRIPTION
**Pull request by gregfromstl**
## PR-Codex overview
This PR updates the SIWE interface to support wallet authentication and login functionalities.

### Detailed summary
- Updated `AuthOption` to include `wallet`
- Updated authentication strategies from "siwe" to "wallet"
- Modified login functions to support wallet authentication
- Added wallet-related functionalities for login with SIWE

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .changeset/grumpy-squids-laugh.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+```ts
+import { inAppWallet, createWallet } from "thirdweb/wallets"
+import { mainnet } from "thirdweb/chains"
+
+const rabby = createWallet("io.rabby");
+const wallet = inAppWallet();
+
+const account = await wallet.connect({
+    client: MY_CLIENT,
+    wallet: rabby,
```
Comment (joaquim-verges) [inline .changeset/grumpy-squids-laugh.md:15]: I would still add strategy: "wallet" 

Seems more consistent with the rest no?

File: packages/thirdweb/src/wallets/in-app/core/authentication/types.ts [hunk 2]
```diff
@@ -43,8 +45,8 @@ export type SingleStepAuthArgsType =
     }
   | {
       strategy: "siwe";
-      walletId: InjectedSupportedWalletIds;
-      chainId: number;
+      wallet: Wallet<InjectedSupportedWalletIds>;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/in-app/core/authentication/types.ts:48]: Does it only work with injected wallets?

File: .changeset/grumpy-squids-laugh.md (hunk 1 above)
Comment (gregfromstl) [reply .changeset/grumpy-squids-laugh.md:15]: Gah yes that's required I just keep forgetting

File: packages/thirdweb/src/wallets/in-app/core/authentication/types.ts (hunk 2 above)
Comment (gregfromstl) [reply packages/thirdweb/src/wallets/in-app/core/authentication/types.ts:48]: Hmm, I guess theoretically you could use any. I'm not sure why you would but I guess you can

Comment (joaquim-verges) [reply packages/thirdweb/src/wallets/in-app/core/authentication/types.ts]: remember the wallet adapter use case

File: .changeset/grumpy-squids-laugh.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+```ts
+import { inAppWallet, createWallet } from "thirdweb/wallets"
+import { mainnet } from "thirdweb/chains"
+
+const rabby = createWallet("io.rabby");
+const wallet = inAppWallet();
+
+const account = await wallet.connect({
+    client: MY_CLIENT,
+    strategy: "siwe",
```
Comment (joaquim-verges) [inline .changeset/grumpy-squids-laugh.md:15]: Change the code example?

**Review by joaquim-verges — APPROVED** [review]

File: .changeset/grumpy-squids-laugh.md (hunk 3 above)
Comment (joaquim-verges) [inline .changeset/grumpy-squids-laugh.md:15]:
```suggestion
    strategy: "wallet",
```

File: packages/thirdweb/src/wallets/in-app/core/authentication/getLoginPath.ts [hunk 4]
```diff
@@ -3,6 +3,15 @@ import { getThirdwebBaseUrl } from "../../../../utils/domains.js";
 import type { AuthOption } from "../../../../wallets/types.js";
 import type { Ecosystem } from "../../web/types.js";
 
+const getLoginOptionRoute = (option: AuthOption) => {
+  switch (option) {
+    case "wallet":
+      return "siwe";
+    default:
+      return option;
```
[the remark this answers is not in the document]
Comment (gregfromstl) [reply packages/thirdweb/src/wallets/in-app/core/authentication/getLoginPath.ts:11]: fair

**Pull request merged**